test-lib.sh

```diff
@@ -1069,4 +1054,34 @@ ct_test_app_dockerfile() {
   return $ret
 }
 
+
+# ct_run_tests_from_testset
+# -----------------------------
+# Runs all tests in $TEST_SET, prints result to
+# the $TEST_SUMMARY variable
+# Argument: app_name - application name to log
+# Uses: $TEST_SET - set of test cases to run
+# Uses: $TEST_SUMMARY - variable for storing test results
+# Uses: $IMAGE_NAME - name of the image being tested
+ct_run_tests_from_testset() {
+  local app_name="$1"
+  local testsuite_result=0
+  for test_case in $TEST_SET; do
+    info "Running test $test_case ... "
+    $test_case
```

Comment:
Move the calling function after variable local definition.

Also add after calling `$test_case`
```bash
check_result
```
like it is used here https://github.com/sclorg/s2i-perl-container/blob/master/test/run#L369

Reply:
Well, the only line which is needed from the `check_result` is the logging one, and IMHO clearer solution of its placement, is where it currently is in this PR. 
```
      test_msg="[FAILED]"
      testsuite_result=1
      echo "Test '${test_case}' for image '${IMAGE_NAME}' FAILED (exit code: ${test_result})"
```
Moreover, the `check_result` isn't owned by container-common-scripts, but is kind of specific to every container  (even though its implementation is quite similar through the sclorg codebase, it is not always the same).
The main function of `check_result` is IMHO to set the `$TESTCASE_RESULT` __global__ variable to 1, if the ret_val is non zero. However, in my opinion, this variable shouldn't be global, but should be in the local scope of `ct_run_tests_from_testset` function, as the only result, which really interests us, is the result [0/1] of the whole test suite.
What do you think?

Reply:
The I would have an idea. Move check_result into `container-common-scripts` as well.
Let's make the whole functionality in `container-common-scripts` repo. The rest of containers will share it and we will remove duplicates. Good catch and approach in the discussion.
the chec_result as soon as after `$test_case` is used in case of `test_case` itself does not check in the function check_result. We need the return value.

Reply:
Ok, I will move the check result to the `test-lib.sh`.